src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/EventService.java

```diff
@@ -430,6 +419,59 @@ public void scheduleEvent(final Event event) throws CTPException {
     }
   }
 
+  /** Get all the scheduled events and send them to action to be acted on. */
+  public void processEvents() {
+    List<Event> eventList = eventRepository.findByStatus(EventDTO.Status.SCHEDULED);
+    log.info("Found [" + eventList.size() + "] events in the SCHEDULED state");
+    for (Event event : eventList) {
+      CollectionExercise exercise = event.getCollectionExercise();
+      boolean isExerciseActive = isCollectionExerciseActive(exercise);
+      boolean isEventInThePast = event.getTimestamp().before(Timestamp.from(Instant.now()));
+      if (isExerciseActive && isEventInThePast) {
+        log.with("id", event.getId()).with("tag", event.getTag()).info("Processing event");
+
+        // If the event is go_live we need to transition the state of the collection exercise
+        Tag tag = EventService.Tag.valueOf(event.getTag());
+        if (tag == EventService.Tag.go_live) {
+          try {
+            collectionExerciseService.transitionCollectionExercise(
+                event.getCollectionExercise(),
+                CollectionExerciseDTO.CollectionExerciseEvent.GO_LIVE);
+            log.with("collection_exercise_id", event.getCollectionExercise().getId())
+                .info("Set collection exercise to LIVE state");
+          } catch (CTPException e) {
+            log.with("collection_exercise_id", event.getCollectionExercise().getId())
+                .error("Failed to set collection exercise to LIVE state", e);
+          }
+        }
+
+        if (tag.isActionable()) {
+          log.with("tag", event.getTag()).info("Event is actionable, beginning processing");
+          // Hard code response until endpoint exists.
+          // boolean success = actionSvcClient.processEvent(event.getTag(), exercise.getId())
+          if (true) {
+            log.info("Event processing succeeded, setting to PROCESSED state");
+            event.setStatus(EventDTO.Status.PROCESSED);
+            event.setMessageSent(Timestamp.from(Instant.now()));
+          } else {
+            log.error("Event processing failed, setting to FAILED state");
```

Comment:
not sure if its not actionable we should add failed to it

Reply:
It's only failed if the call to `/process-event` in action fails.  If an event isn't actionable then it's reasonable to set it to processed because theres nothing we need to do, it hasn't failed because it never started

Reply:
👍